rdmo/projects/assets/js/interview/components/main/widget/RangeInput.js

```diff
@@ -1,34 +1,51 @@
-import React, { useState, useEffect } from 'react'
+import React, { useEffect, useLayoutEffect, useRef, useState } from 'react'
 import PropTypes from 'prop-types'
 import { useDebouncedCallback } from 'use-debounce'
 import classNames from 'classnames'
-import { isEmpty } from 'lodash'
+import { isNil, toString } from 'lodash'
 
 import { isDefaultValue } from '../../../utils/value'
 
 import Unit from './common/Unit'
 
 const RangeInput = ({ question, value, disabled, updateValue, buttons }) => {
+  const ref = useRef(null)
   const [inputValue, setInputValue] = useState('')
-  useEffect(() => {setInputValue(value.text)}, [value.text])
+
+  useEffect(() => setInputValue(value.text), [value.text])
+
+  useLayoutEffect(() => {
+    if (ref.current) {
+      const unitElement = ref.current.nextElementSibling
+
+      const unitWidth = (
+        isNil(unitElement.lastChild) ? unitElement.lastChild.offsetWidth : 0
+      ) + (
+        isNil(question.maximum) ? 30 : toString(question.maximum).length * 10
+      )
+
+      unitElement.style.flex = `0 0 ${unitWidth}px`
+    }
+  }, [question])
 
   const handleChange = useDebouncedCallback((value, text) => {
     updateValue(value, { text, unit: question.unit, value_type: question.value_type })
   }, 500)
 
   const classnames = classNames({
     'interview-input': true,
-    'range': true,
+    'range-input': true,
     'default': isDefaultValue(question, value)
   })
 
   return (
     <div className={classnames}>
       <input
+        ref={ref}
         type="range"
-        min={isEmpty(question.minimum) ? '0' : question.minimum}
-        max={isEmpty(question.maximum) ? '100' : question.maximum}
-        step={isEmpty(question.step) ? '1' : question.step}
+        min={isNil(question.minimum) ? '0' : question.minimum}
+        max={isNil(question.maximum) ? '100' : question.maximum}
+        step={isNil(question.step) ? '1' : question.step}
         disabled={disabled}
         value={inputValue}
```

Comment:
I get a type warning for the `Unit` 
```js
Warning: Failed prop type: Invalid prop `inputValue` of type `string` supplied to `Unit`, expected `number`.
Unit@webpack://rdmo/./rdmo/projects/assets/js/interview/components/main/widget/common/Unit.js?:15:14
RangeInput@webpack://rdmo/./rdmo/projects/assets/js/interview/components/main/widget/RangeInput.js?:30:18
div
```

Reply:
Interestingly, I could not reproduce it. Maybe something with the browser. I will just change it to:

```
Unit.propTypes = {
  unit: PropTypes.string,
  inputValue: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
}
```